### ISSUE TITLE
Add a genicam feature control property to aravis gstreamer plugin

### ISSUE THREAD

**Issue by EmmanuelP** [AravisProject/aravis#319]
Sometimes, using the gstreamer plugin, it is necessary to set some device feature in order to make it work properly.

A possible implementation would be to add a `features` property, that takes a list of feature/value pairs, similar to the parameters of the `control` command of arv-tool.

Then we could write:

```
gst-launch-1.0 aravissrc features="DeviceLinkThroughputLimit=185000000 ShutterMode=Rolling" ! videoconvert ! autovideosink
```

**Comment by mcclauba** [comment]
A very good idea.  Just need to be sure that a few features are required to be in the correct mode for their values to be manually set.

**Comment by barzan-hayati** [comment]
> Sometimes, using the gstreamer plugin, it is necessary to set some device feature in order to make it work properly.
> 
> A possible implementation would be to add a `features` property, that takes a list of feature/value pairs, similar to the parameters of the `control` command of arv-tool.
> 
> Then we could write:
> 
> ```
> gst-launch-1.0 aravissrc features="DeviceLinkThroughputLimit=185000000 ShutterMode=Rolling" ! videoconvert ! autovideosink
> ```

Hi @EmmanuelP . `Features` command does not work for me.
`gst-launch-1.0 aravissrc features="DeviceLinkThroughputLimit=185000000 ShutterMode=Rolling" ! videoconvert ! autovideosink`


`WARNING: erroneous pipeline: no property "features" in element "aravis0"
`

and by adding aravis source address:

` gst-launch-1.0 -e --gst-plugin-path=/usr/local/lib/ aravissrc features="DeviceLinkThroughputLimit=185000000 ShutterMode=Rolling" ! videoconvert ! autovideosink`


`WARNING: erroneous pipeline: no property "features" in element "aravis0"
`

**Comment by EmmanuelP** [comment]
@barzan-hayati it is not implemented yet...

**Comment by EmmanuelP** [comment]
Implemented in a755517e44b1190379e151c7403dc861ce88bedf

**Comment by nrstrifork** [comment]
Hi @EmmanuelP,

Did you document all genicam features, or are you using the same feature key/val naming as genicam?

**Comment by EmmanuelP** [comment]
Hi,

> Did you document all genicam features, or are you using the same feature key/val naming as genicam?

I fail to understand what you mean. Could you rephrase your question ?

**Comment by EmmanuelP** [comment]
You can set any feature available on your camera, using the same syntax as the arv-tool utility. You can list every available features using `arv-tool-0.8 features`.

**Comment by nrstrifork** [comment]
> You can set any feature available on your camera, using the same syntax as the `control` command of `arv-tool`. You can list every available features using `arv-tool-0.8 features`.

Thank you :)
